src/modules/client/components/customers/billing/CreditNoteEditionModal.vue

```diff
@@ -117,9 +122,13 @@ export default {
     submit () {
       this.$emit('submit');
     },
-    getEvents (value) {
+    getEvents (event, value) {
+      this.update(event, value);
       this.$emit('get-events', value);
     },
+    update (event, path) {
+      this.$emit('update:editedCreditNote', set({ ...this.editedCreditNote }, path, event));
+    },
```

Comment:
Il faudra peut-être qu'on homogénise ça : 
- cette fonction update est utilisée lorsqu'on update un champ profond (`editedCreditNote.identity.firstName` par exemple)
- on utilise `this.$emit('update:editedCreditNote', { ...this.editedCreditNote, [prop]: event });` lorsqu'on a pas besoin d'accéder à un champ profond

Qu'en pensez-vous ?

Reply:
je ne comprends pas ton messsage ^^

Reply:
Pour le refacto vue/no-mutating props, on utilise deux fonctions différentes pour l'update : 
La première est : 
```
    update (event, prop) {
      this.$emit('update:editedCreditNote', { ...this.editedCreditNote, [prop]: event });
    },
```
On l'utilise lorsque tout les champs à modifier son directement accessible (variable.champ) : 
`editedCreditNote.date`, `editedCreditNote.name`, ....
c'est le cas pour SmsSendingModal par exemple

La seconde est :
```
    update (event, path) {
      this.$emit('update:editedCreditNote', set({ ...this.editedCreditNote }, path, event));
    },
```
On l'utilise lorsqu'il y a besoin de modifier ce que j'appelle un champ profond (variable.champ.champ-profond) : 
editedCreditNote.date.startDate, editedCreditNote.name.firstName, ...
c'est le cas pour CoachCreationModal par exemple

Je me demandais donc si c'était utile d'avoir ces deux fonctions, sachant que la deuxième fonctionne dans tous les cas (champ directement accessible et champ profond).

Reply:
je pense qu'il ne faut pas avoir les 2 méthodes dans un même fichier, puisque la deuxième fonctionne très bien dans le premier cas.
Après je me dis qu'on a d'autres améliorations de code plus prios que repasser sur toutes les modales pour homogénéiser ça

donc si a l'occas, au cours d'un développement on repasse dessus, pourquoi pas mais je ne suis pass tres chaud pour repartir sur une série de tiketec pour changer ca

Reply:
totalement d'accord